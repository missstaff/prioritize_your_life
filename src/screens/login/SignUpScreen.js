import React, { useEffect, useState } from "react";
import { ScrollView, Text, TouchableOpacity, View } from "react-native";
import { useNavigation } from "@react-navigation/native";
import AppTextInput from "../../components/ui/AppTextInput";
import { getFireApp } from "../../../getFireApp";
import { displayErrorToast, validateTextInput } from "../../utility/utilities";
import { APP_COLORS } from "../../utility/constants";
import { styles } from "./Styles";



const SignUpScreen = () => {

    const firebase = getFireApp();
    const navigation = useNavigation();

    const [submitIsDisabled, setSubmitIsDisabled] = useState(true);
    const [emailIsValid, setEmailIsValid] = useState(false);
    const [passwordIsValid, setPasswordIsValid] = useState(false);
    const [confirmPasswordIsValid, setConfirmPasswordIsValid] = useState(false);
    const [userNameIsValid, setUserNameIsValid] = useState(false);
    const [state, setState] = useState({
        email: "",
        password: "",
        confirmPassword: "",
        username: "",
    });


    const onPressForgotPassword = () => {
        navigation.navigate("PasswordResetScreen");
    };

    const onPressSubmit = async () => {
        try {
            await firebase.auth().createUserWithEmailAndPassword(state.email, state.password);
        } catch (error) {
            console.log(`Error: ${error.message}\n${error.stack}`);
            displayErrorToast("Failed to create account.", "Please try again.");
        }
    };

    useEffect(() => {
        if (userNameIsValid && emailIsValid && passwordIsValid && confirmPasswordIsValid) {
            setSubmitIsDisabled(false);
        } else {
            setSubmitIsDisabled(true);
        }
    }, [userNameIsValid, emailIsValid, passwordIsValid, confirmPasswordIsValid]);


    return (
        <ScrollView
            contentContainerStyle={{
                alignItems: "center",
                backgroundColor: APP_COLORS.primary,
                flexGrow: 1,
                justifyContent: "center",
            }}>
            <Text style={styles.title}>SignUp</Text>
            <View style={styles.inputView}>
                <AppTextInput
                    style={styles.inputText}
                    placeholder="Username"
                    onChangeText={(text) => setState({ ...state, username: text.trim() })}
                    onBlur={() =>
                        validateTextInput(
                            {
                                callback: setUserNameIsValid,
                                condition: state.username.length > 1,
                                errorText1: "Username must be at least 2 letters.",
                                errorText2: "Please try again.",
                                type: 1,
                            }
                        )
                    }
                />
            </View>
            <View style={styles.inputView}>
                <AppTextInput
                    style={styles.inputText}
                    placeholder="Email"
                    onChangeText={(text) => setState({ ...state, email: text.trim() })}
                    onBlur={() =>
                        validateTextInput(
                            {
                                callback: setEmailIsValid,
                                condition: state.email,
                                errorText1: "A valid email is required.",
                                errorText2: "Please try again.",
                                type: 2,
                            }
                        )
                    }
                />
            </View>
            <View style={styles.inputView}>
                <AppTextInput
                    style={styles.inputText}
                    placeholder="Password"
                    onChangeText={(text) => setState({ ...state, password: text.trim() })}
                    onBlur={() =>
                        validateTextInput(
                            {
                                callback: setPasswordIsValid,
                                condition: state.password,
                                errorText1: "Password must be 7 to 25 characters and contain at least one uppercase letter, one lowercase letter, one number, and one special character.",
                                errorText2: "Please try again.",
                                type: 3,
                            }
                        )
                    }
                />
            </View>
            <View style={styles.inputView}>
                <AppTextInput
                    style={styles.inputText}
                    placeholder="Confirm Password"
                    onChangeText={(text) => setState({ ...state, confirmPassword: text.trim() })}
                    onBlur={() =>
                        validateTextInput(
                            {
                                callback: setConfirmPasswordIsValid,
                                condition: state.confirmPassword === state.password,
                                errorText1: "Passwords must match.",
                                errorText2: "Please try again.",
                                type: 4,
                            }
                        )
                    }
                />
            </View>
            <TouchableOpacity
                onPress={onPressForgotPassword}>
                <Text style={styles.forgotAndSignUpText}>Forgot Password?</Text>
            </TouchableOpacity>
            <TouchableOpacity
                disabled={submitIsDisabled}
                onPress={onPressSubmit}
                style={styles.loginBtn}>
                <Text style={styles.loginText}>Submit</Text>
            </TouchableOpacity>
        </ScrollView>
    );
};

export default SignUpScreen;