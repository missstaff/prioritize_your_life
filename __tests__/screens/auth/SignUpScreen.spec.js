import renderer from "react-test-renderer";
import SignUpScreen from "../../../src/screens/auth/SignUpScreen";


describe("SignUpScreen", () => {

  afterAll(() => {
    jest.clearAllMocks();
  });

  it("SignUpScreen should render correctly", () => {
    const tree = renderer.create(<SignUpScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});