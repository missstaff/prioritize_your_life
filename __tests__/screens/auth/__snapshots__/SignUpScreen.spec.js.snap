// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpScreen SignUpScreen should render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "backgroundColor": "#4FD3DA",
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#FFC857",
          "fontSize": 107.14285714285714,
          "fontWeight": "bold",
          "marginBottom": 78.47058823529412,
        }
      }
    >
      SignUp
    </Text>
    <View
      style={
        {
          "backgroundColor": "#3AB4BA",
          "borderRadius": 39.285714285714285,
          "height": 107.14285714285714,
          "justifyContent": "center",
          "marginBottom": 39.23529411764706,
          "padding": 42.857142857142854,
          "width": "80%",
        }
      }
    >
      <View>
        <TextInput
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="#003f5c"
          style={
            {
              "color": "#FFFFFF",
              "height": 107.14285714285714,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#3AB4BA",
          "borderRadius": 39.285714285714285,
          "height": 107.14285714285714,
          "justifyContent": "center",
          "marginBottom": 39.23529411764706,
          "padding": 42.857142857142854,
          "width": "80%",
        }
      }
    >
      <View>
        <TextInput
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="#003f5c"
          style={
            {
              "color": "#FFFFFF",
              "height": 107.14285714285714,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#3AB4BA",
          "borderRadius": 39.285714285714285,
          "height": 107.14285714285714,
          "justifyContent": "center",
          "marginBottom": 39.23529411764706,
          "padding": 42.857142857142854,
          "width": "80%",
        }
      }
    >
      <View>
        <TextInput
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="#003f5c"
          style={
            {
              "color": "#FFFFFF",
              "height": 107.14285714285714,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#3AB4BA",
          "borderRadius": 39.285714285714285,
          "height": 107.14285714285714,
          "justifyContent": "center",
          "marginBottom": 39.23529411764706,
          "padding": 42.857142857142854,
          "width": "80%",
        }
      }
    >
      <View>
        <TextInput
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Confirm Password"
          placeholderTextColor="#003f5c"
          style={
            {
              "color": "#FFFFFF",
              "height": 107.14285714285714,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="Forgot password link"
      accessibilityRole="link"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 23.57142857142857,
          }
        }
      >
        Forgot Password?
      </Text>
    </View>
    <View
      accessibilityLabel="Submit button"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFC857",
          "borderRadius": 39.285714285714285,
          "height": 107.14285714285714,
          "justifyContent": "center",
          "marginBottom": 19.61764705882353,
          "marginTop": 78.47058823529412,
          "opacity": 1,
          "width": "80%",
        }
      }
    >
      <Text>
        Submit
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
